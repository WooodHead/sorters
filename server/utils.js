function prepare(o) {
    if (o && o._id) {
        o._id = o._id.toString()
    }
    return o
}

async function findOne(Collection, ...args) {
    return prepare(await Collection.findOne(...args))
}

async function find(Collection, ...args) {
    return (await Collection.find(...args).toArray()).map(prepare)
}

module.exports = {
    prepare,
    findOne,
    find,
}