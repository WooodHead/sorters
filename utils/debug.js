
export function log(arg) {
    console.log(arg)
    return arg
}
