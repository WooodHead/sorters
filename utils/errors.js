export const errorMessage = (e) => {
    if (e.graphQLErrors && e.graphQLErrors.length > 0) {
        return e.graphQLErrors[0].message
    } else {
        return e.message
    }
}