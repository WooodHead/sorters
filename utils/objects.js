
function toObject(arr) {
    const res = {}
    for (const [key, value] of arr) {
        res[key] = value
    }
    return res
}

function objectMap(object, keyMap = key => key, valueMap = value => value) {
    return Object.keys(object).reduce((o, key) => {
        o[keyMap(key)] = valueMap(object[key])
        return o
    }, {})
}

module.exports = {
    toObject,
    objectMap
}
