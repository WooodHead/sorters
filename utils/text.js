export const replaceAll = (string, search, replacement) => {
  if (typeof string === "string") {
    return string.replace(new RegExp(search, 'g'), replacement)
  } else {
    return string
  }
}

export const teaserAndInfo = (content, min, opt, max) => {
	if (content) {
		if (content.length <= max) {
			return {
				cut: false,
				teaser: content
			};
		}
		const words = content.split(" ");
		let teaser = "";
		for (let i=0; i<words.length; i++) {
			const candidate = words[i];
			let wouldbelength = teaser.length + 1 + candidate.length + 3;
			if (wouldbelength >= opt) {
				// We're done
				if (wouldbelength > max) {
					if (teaser.length + 3 < min) {
						// cut the word
						return {
							teaser: teaser + " " + candidate.substring(0, opt - teaser.length - 1) + "...",
							cut: true,
						};
					} else {
						return {
							teaser: teaser + "...",
							cut: true,
						};
					}
				} else {
					return {
						teaser: teaser + " " + candidate + "...",
						cut: true,
					};
				}
			}
			teaser += " " + candidate;
		}
		return {
			teaser: teaser + "...",
			cut: true,
		};
	} else {
		return {
			teaser: "",
			cut: false,
		};
	}
};
export const teaser = (content, min, opt, max) => {
	const teaser = teaserAndInfo(content, min, opt, max);
	return teaser.teaser;
}
