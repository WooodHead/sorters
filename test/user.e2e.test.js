import cheerio from 'cheerio'
import pretty from 'pretty'
import {ObjectID} from 'mongodb'
import {setup, teardown} from './setup'
import {generateAndLogUser, setUserData} from './fixtures'
import {request} from './utils'

let db
let browser

beforeAll(async () => {
    const res = await setup()
    db = res.db
    browser = res.browser
})

afterEach(async () => {
    await db.dropDatabase()    
})

afterAll(async () => {
    await teardown({db, browser})
})

describe('user', () => {
    it('displays', async () => {
        const browserPage = await browser.createPage()

        await generateAndLogUser(browserPage)
        await setUserData(browserPage)
        
        const status = await browserPage.open(`http://localhost:3000/u/test`)
        expect(status).toBe('success')

        const text = await browserPage.property('content')
        const $ = cheerio.load(text)
        const page = $('#__next')
        const html = pretty(page.html())
        expect(html).toMatchSnapshot()
    })

    it('displays goals', async () => {
        const browserPage = await browser.createPage()

        await generateAndLogUser(browserPage)
        await setUserData(browserPage)
        
        const status = await browserPage.open(`http://localhost:3000/u/test/goals`)
        expect(status).toBe('success')

        const text = await browserPage.property('content')
        const $ = cheerio.load(text)
        const page = $('#__next')
        const html = pretty(page.html())
        expect(html).toMatchSnapshot()
    })

    it('displays journal', async () => {
        const browserPage = await browser.createPage()

        await generateAndLogUser(browserPage)
        await setUserData(browserPage)
        await request(browserPage, `http://localhost:3000/graphql`, 'POST', {
            query: `
                mutation($entry: NewEntryInput!) {
                    createEntry(entry: $entry) {
                        _id
                    }
                }
            `,
            variables: {
                entry: {
                    title: 'Hello World',
                    url: 'http://www.example.com',
                    description: 'Lorem ipsum',
                    goalTitles: ['Goal', 'Goal - doing'],
                },
            },
        })
        
        const status = await browserPage.open(`http://localhost:3000/u/test/journal`)
        expect(status).toBe('success')

        const text = await browserPage.property('content')
        const $ = cheerio.load(text)
        const page = $('#__next')
        const html = pretty(page.html())
        expect(html).toMatchSnapshot()
    })

    it('displays profile', async () => {
        const browserPage = await browser.createPage()

        await generateAndLogUser(browserPage)
        await setUserData(browserPage)
        
        const status = await browserPage.open(`http://localhost:3000/u/test/profile`)
        expect(status).toBe('success')

        const text = await browserPage.property('content')
        const $ = cheerio.load(text)
        const page = $('#__next')
        const html = pretty(page.html())
        expect(html).toMatchSnapshot()
    })

    it('displays reads', async () => {
        const browserPage = await browser.createPage()

        await generateAndLogUser(browserPage)
        await setUserData(browserPage)
        
        const status = await browserPage.open(`http://localhost:3000/u/test/reads`)
        expect(status).toBe('success')

        const text = await browserPage.property('content')
        const $ = cheerio.load(text)
        const page = $('#__next')
        const html = pretty(page.html())
        expect(html).toMatchSnapshot()
    })

})