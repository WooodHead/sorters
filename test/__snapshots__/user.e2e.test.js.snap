// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user displays 1`] = `
"<div data-reactroot=\\"\\">
  <div>
    <div style=\\"padding-top:50px;padding-bottom:20px\\">
      <nav class=\\"navbar navbar-inverse navbar-fixed-top\\">
        <div class=\\"container\\">
          <div class=\\"navbar-header\\">
            <button type=\\"button\\" class=\\"navbar-toggle collapsed\\" data-toggle=\\"collapse\\" data-target=\\"#navbar\\" aria-expanded=\\"false\\" aria-controls=\\"navbar\\"><span class=\\"sr-only\\">Toggle navigation</span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span></button>
            <a class=\\"navbar-brand\\" href=\\"/\\">Sorters Club</a>
          </div>
          <div id=\\"navbar\\" class=\\"collapse navbar-collapse\\">
            <ul class=\\"nav navbar-nav navbar-right\\">
              <li>
                <a href=\\"/\\">Home</a>
              </li>
              <li>
                <a href=\\"/about\\">About</a>
              </li>
              <li>
                <a href=\\"/users\\">Sorters</a>
              </li>
              <li class=\\"dropdownundefined\\">
                <a href=\\"#\\" class=\\"dropdown-toggle\\" data-toggle=\\"dropdown\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"true\\">Hello, test <span class=\\"caret\\"></span></a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
    <div class=\\"container\\">
      <div>
        <h1>Test User</h1>
        <div class=\\"row\\">
          <div class=\\"col-xs-6 col-sm-3 col-md-2\\">
            <img alt=\\"Gravatar for undefined\\" style=\\"margin-bottom:1.5rem;width:100%;height:auto;margin-left:auto;margin-right:auto;display:block\\" src=\\"//www.gravatar.com/avatar/23463b99b62a72f26ed677cc556c44e8?d=retro&amp;r=g&amp;s=200\\" srcset=\\"//www.gravatar.com/avatar/23463b99b62a72f26ed677cc556c44e8?d=retro&amp;r=g&amp;s=400 2x\\"
              height=\\"200\\" width=\\"200\\" class=\\"react-gravatar\\">
          </div>
          <div class=\\"col-xs-12 col-sm-9 col-md-10\\">
            <p>
              <a href=\\"/u/test\\">/u/ test</a>
            </p>
            <div>
              <div class=\\"markdown\\">
                <p>Test user about</p>
              </div>
            </div>
            <button class=\\"btn btn-primary\\">Send message</button>
          </div>
        </div>
        <ul class=\\"nav nav-tabs\\">
          <li role=\\"presentation\\" class=\\"active\\">
            <a href=\\"/u/test\\">&#x26CF; Activity</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/u/test/profile\\">&#x1F464; Profile</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/u/test/goals\\">&#x25CE; Goals</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/u/test/journal\\">&#x270E; Journal</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/u/test/reads\\">&#x1F4D6; Books</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/u/test/topics\\">&#x1F4A1; Topics</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/u/test/essays\\">&#x270E; Essays</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/u/test/speeches\\">&#x1F444; Speeches</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/u/test/conversations\\">&#x1F5E9; Conversations</a>
          </li>
        </ul>
      </div>
      <h2>Activity</h2>
      <div style=\\"margin-bottom:1.5rem\\">&#x1F4D6; Wants to read
        <a href=\\"/read/<id>\\">Read - read</a>.</div>
      <div style=\\"margin-bottom:1.5rem\\">&#x1F4D6; Wants to read
        <a href=\\"/read/<id>\\">Read - reading</a>.</div>
      <div style=\\"margin-bottom:1.5rem\\">&#x1F4D6; Wants to read
        <a href=\\"/read/<id>\\">Read</a>.</div>
      <div style=\\"margin-bottom:1.5rem\\">&#x25CE; Wants to achieve
        <a href=\\"/goal/<id>\\">Goal - done</a>.</div>
      <div style=\\"margin-bottom:1.5rem\\">&#x25CE; Wants to achieve
        <a href=\\"/goal/<id>\\">Goal - doing</a>.</div>
      <div style=\\"margin-bottom:1.5rem\\">&#x25CE; Wants to achieve
        <a href=\\"/goal/<id>\\">Goal</a>.</div>
      <div style=\\"margin-bottom:1.5rem\\">&#x1F4D6; Updated
        <a href=\\"/u/test/goals\\">goals description</a>.</div>
      <div style=\\"margin-bottom:1.5rem\\">&#x1F4D6; Updated
        <a href=\\"/u/test/reads\\">reading list description</a>.</div>
      <div style=\\"margin-bottom:1.5rem\\">&#x1F464; Updated
        <a href=\\"/u/test/profile\\">profile</a>.</div>
      <div style=\\"margin-bottom:1.5rem\\">&#x2714; Completed
        <a href=\\"/u/test/profile\\">Past authoring</a>.</div>
      <div style=\\"margin-bottom:1.5rem\\">&#x2714; Completed
        <a href=\\"/u/test/profile\\">Present virtues authoring</a>.</div>
      <div style=\\"margin-bottom:1.5rem\\">&#x2714; Completed
        <a href=\\"/u/test/profile\\">Present faults authoring</a>.</div>
      <div style=\\"margin-bottom:1.5rem\\">&#x2714; Completed
        <a href=\\"/u/test/profile\\">Future authoring</a>.</div>
      <div style=\\"margin-bottom:1.5rem\\">&#x2714; Completed
        <a href=\\"/u/test/profile\\">Understand myself</a>.</div>
    </div>
    <footer>
      <div class=\\"container\\">
        <hr>
        <p>Brought to you with &#x2764; by
          <a href=\\"/about\\">Nick Redmark</a>. Support this platform on
          <a href=\\"https://www.patreon.com/nickredmark\\" target=\\"_blank\\">Patreon</a>.</p>
      </div>
    </footer>
  </div>
</div>"
`;

exports[`user displays goals 1`] = `
"<div data-reactroot=\\"\\">
  <div>
    <div style=\\"padding-top:50px;padding-bottom:20px\\">
      <nav class=\\"navbar navbar-inverse navbar-fixed-top\\">
        <div class=\\"container\\">
          <div class=\\"navbar-header\\">
            <button type=\\"button\\" class=\\"navbar-toggle collapsed\\" data-toggle=\\"collapse\\" data-target=\\"#navbar\\" aria-expanded=\\"false\\" aria-controls=\\"navbar\\"><span class=\\"sr-only\\">Toggle navigation</span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span></button>
            <a class=\\"navbar-brand\\" href=\\"/\\">Sorters Club</a>
          </div>
          <div id=\\"navbar\\" class=\\"collapse navbar-collapse\\">
            <ul class=\\"nav navbar-nav navbar-right\\">
              <li>
                <a href=\\"/\\">Home</a>
              </li>
              <li>
                <a href=\\"/about\\">About</a>
              </li>
              <li>
                <a href=\\"/users\\">Sorters</a>
              </li>
              <li class=\\"dropdownundefined\\">
                <a href=\\"#\\" class=\\"dropdown-toggle\\" data-toggle=\\"dropdown\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"true\\">Hello, test <span class=\\"caret\\"></span></a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
    <div class=\\"container\\">
      <div>
        <div>
          <h1>Test User</h1>
          <div class=\\"row\\">
            <div class=\\"col-xs-6 col-sm-3 col-md-2\\">
              <img alt=\\"Gravatar for undefined\\" style=\\"margin-bottom:1.5rem;width:100%;height:auto;margin-left:auto;margin-right:auto;display:block\\" src=\\"//www.gravatar.com/avatar/23463b99b62a72f26ed677cc556c44e8?d=retro&amp;r=g&amp;s=200\\" srcset=\\"//www.gravatar.com/avatar/23463b99b62a72f26ed677cc556c44e8?d=retro&amp;r=g&amp;s=400 2x\\"
                height=\\"200\\" width=\\"200\\" class=\\"react-gravatar\\">
            </div>
            <div class=\\"col-xs-12 col-sm-9 col-md-10\\">
              <p>
                <a href=\\"/u/test\\">/u/ test</a>
              </p>
              <div>
                <div class=\\"markdown\\">
                  <p>Test user about</p>
                </div>
              </div>
              <button class=\\"btn btn-primary\\">Send message</button>
            </div>
          </div>
          <ul class=\\"nav nav-tabs\\">
            <li role=\\"presentation\\">
              <a href=\\"/u/test\\">&#x26CF; Activity</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/profile\\">&#x1F464; Profile</a>
            </li>
            <li role=\\"presentation\\" class=\\"active\\">
              <a href=\\"/u/test/goals\\">&#x25CE; Goals</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/journal\\">&#x270E; Journal</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/reads\\">&#x1F4D6; Books</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/topics\\">&#x1F4A1; Topics</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/essays\\">&#x270E; Essays</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/speeches\\">&#x1F444; Speeches</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/conversations\\">&#x1F5E9; Conversations</a>
            </li>
          </ul>
        </div>
        <h2>Goals</h2>
        <div class=\\"markdown\\">
          <p>Test user goals</p>
        </div>
        <ul>
          <li>
            <a href=\\"/goal/<id>\\">Goal</a>
            <ul></ul>
          </li>
          <li>
            <a href=\\"/goal/<id>\\">Goal - doing</a>
            <ul></ul>
          </li>
          <li>
            <a href=\\"/goal/<id>\\">Goal - done</a>
            <ul></ul>
          </li>
        </ul>
      </div>
    </div>
    <footer>
      <div class=\\"container\\">
        <hr>
        <p>Brought to you with &#x2764; by
          <a href=\\"/about\\">Nick Redmark</a>. Support this platform on
          <a href=\\"https://www.patreon.com/nickredmark\\" target=\\"_blank\\">Patreon</a>.</p>
      </div>
    </footer>
  </div>
</div>"
`;

exports[`user displays journal 1`] = `
"<div data-reactroot=\\"\\">
  <div>
    <div style=\\"padding-top:50px;padding-bottom:20px\\">
      <nav class=\\"navbar navbar-inverse navbar-fixed-top\\">
        <div class=\\"container\\">
          <div class=\\"navbar-header\\">
            <button type=\\"button\\" class=\\"navbar-toggle collapsed\\" data-toggle=\\"collapse\\" data-target=\\"#navbar\\" aria-expanded=\\"false\\" aria-controls=\\"navbar\\"><span class=\\"sr-only\\">Toggle navigation</span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span></button>
            <a class=\\"navbar-brand\\" href=\\"/\\">Sorters Club</a>
          </div>
          <div id=\\"navbar\\" class=\\"collapse navbar-collapse\\">
            <ul class=\\"nav navbar-nav navbar-right\\">
              <li>
                <a href=\\"/\\">Home</a>
              </li>
              <li>
                <a href=\\"/about\\">About</a>
              </li>
              <li>
                <a href=\\"/users\\">Sorters</a>
              </li>
              <li class=\\"dropdownundefined\\">
                <a href=\\"#\\" class=\\"dropdown-toggle\\" data-toggle=\\"dropdown\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"true\\">Hello, test <span class=\\"caret\\"></span></a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
    <div class=\\"container\\">
      <div>
        <div>
          <h1>Test User</h1>
          <div class=\\"row\\">
            <div class=\\"col-xs-6 col-sm-3 col-md-2\\">
              <img alt=\\"Gravatar for undefined\\" style=\\"margin-bottom:1.5rem;width:100%;height:auto;margin-left:auto;margin-right:auto;display:block\\" src=\\"//www.gravatar.com/avatar/23463b99b62a72f26ed677cc556c44e8?d=retro&amp;r=g&amp;s=200\\" srcset=\\"//www.gravatar.com/avatar/23463b99b62a72f26ed677cc556c44e8?d=retro&amp;r=g&amp;s=400 2x\\"
                height=\\"200\\" width=\\"200\\" class=\\"react-gravatar\\">
            </div>
            <div class=\\"col-xs-12 col-sm-9 col-md-10\\">
              <p>
                <a href=\\"/u/test\\">/u/ test</a>
              </p>
              <div>
                <div class=\\"markdown\\">
                  <p>Test user about</p>
                </div>
              </div>
              <button class=\\"btn btn-primary\\">Send message</button>
            </div>
          </div>
          <ul class=\\"nav nav-tabs\\">
            <li role=\\"presentation\\">
              <a href=\\"/u/test\\">&#x26CF; Activity</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/profile\\">&#x1F464; Profile</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/goals\\">&#x25CE; Goals</a>
            </li>
            <li role=\\"presentation\\" class=\\"active\\">
              <a href=\\"/u/test/journal\\">&#x270E; Journal</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/reads\\">&#x1F4D6; Books</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/topics\\">&#x1F4A1; Topics</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/essays\\">&#x270E; Essays</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/speeches\\">&#x1F444; Speeches</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/conversations\\">&#x1F5E9; Conversations</a>
            </li>
          </ul>
        </div>
        <h2>Journal</h2>
        <div style=\\"margin-top:1.5rem;margin-bottom:1.5rem\\">
          <h3>
            <a href=\\"http://www.example.com\\" target=\\"_blank\\">Hello World</a>
          </h3>
          <div class=\\"markdown\\">
            <p>Lorem ipsum</p>
          </div>
          <div>Goals: <span> <a href=\\"/goal/<id>\\">Goal</a></span><span>, <a href=\\"/goal/<id>\\">Goal - doing</a></span><span>, <a href=\\"/goal/<id>\\">Goal - done</a></span></div>
          <a href=\\"/entry/<id>\\">Comments</a>
          <hr>
        </div>
      </div>
    </div>
    <footer>
      <div class=\\"container\\">
        <hr>
        <p>Brought to you with &#x2764; by
          <a href=\\"/about\\">Nick Redmark</a>. Support this platform on
          <a href=\\"https://www.patreon.com/nickredmark\\" target=\\"_blank\\">Patreon</a>.</p>
      </div>
    </footer>
  </div>
</div>"
`;

exports[`user displays profile 1`] = `
"<div data-reactroot=\\"\\">
  <div>
    <div style=\\"padding-top:50px;padding-bottom:20px\\">
      <nav class=\\"navbar navbar-inverse navbar-fixed-top\\">
        <div class=\\"container\\">
          <div class=\\"navbar-header\\">
            <button type=\\"button\\" class=\\"navbar-toggle collapsed\\" data-toggle=\\"collapse\\" data-target=\\"#navbar\\" aria-expanded=\\"false\\" aria-controls=\\"navbar\\"><span class=\\"sr-only\\">Toggle navigation</span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span></button>
            <a class=\\"navbar-brand\\" href=\\"/\\">Sorters Club</a>
          </div>
          <div id=\\"navbar\\" class=\\"collapse navbar-collapse\\">
            <ul class=\\"nav navbar-nav navbar-right\\">
              <li>
                <a href=\\"/\\">Home</a>
              </li>
              <li>
                <a href=\\"/about\\">About</a>
              </li>
              <li>
                <a href=\\"/users\\">Sorters</a>
              </li>
              <li class=\\"dropdownundefined\\">
                <a href=\\"#\\" class=\\"dropdown-toggle\\" data-toggle=\\"dropdown\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"true\\">Hello, test <span class=\\"caret\\"></span></a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
    <div class=\\"container\\">
      <div>
        <div>
          <h1>Test User</h1>
          <div class=\\"row\\">
            <div class=\\"col-xs-6 col-sm-3 col-md-2\\">
              <img alt=\\"Gravatar for undefined\\" style=\\"margin-bottom:1.5rem;width:100%;height:auto;margin-left:auto;margin-right:auto;display:block\\" src=\\"//www.gravatar.com/avatar/23463b99b62a72f26ed677cc556c44e8?d=retro&amp;r=g&amp;s=200\\" srcset=\\"//www.gravatar.com/avatar/23463b99b62a72f26ed677cc556c44e8?d=retro&amp;r=g&amp;s=400 2x\\"
                height=\\"200\\" width=\\"200\\" class=\\"react-gravatar\\">
            </div>
            <div class=\\"col-xs-12 col-sm-9 col-md-10\\">
              <p>
                <a href=\\"/u/test\\">/u/ test</a>
              </p>
              <div>
                <div class=\\"markdown\\">
                  <p>Test user about</p>
                </div>
              </div>
              <button class=\\"btn btn-primary\\">Send message</button>
            </div>
          </div>
          <ul class=\\"nav nav-tabs\\">
            <li role=\\"presentation\\">
              <a href=\\"/u/test\\">&#x26CF; Activity</a>
            </li>
            <li role=\\"presentation\\" class=\\"active\\">
              <a href=\\"/u/test/profile\\">&#x1F464; Profile</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/goals\\">&#x25CE; Goals</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/journal\\">&#x270E; Journal</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/reads\\">&#x1F4D6; Books</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/topics\\">&#x1F4A1; Topics</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/essays\\">&#x270E; Essays</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/speeches\\">&#x1F444; Speeches</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/conversations\\">&#x1F5E9; Conversations</a>
            </li>
          </ul>
        </div>
        <h2>Summary</h2>
        <ul>
          <li>&#x25CE; 3 
            <a href=\\"/u/test/goals\\">goals</a>
          </li>
          <li>&#x1F4D6; 3 
            <a href=\\"/u/test/reads\\">books in reading list</a>
          </li>
          <li>A man</li>
          <li>City: Zurich , Switzerland</li>
        </ul>
        <div>
          <h2>Links</h2>
          <ul>
            <li>
              <label style=\\"width:100px;margin-right:1.5rem\\">Website</label>
              <a href=\\"http://website\\">http://website</a>
            </li>
            <li>
              <label style=\\"width:100px;margin-right:1.5rem\\">Blog</label>
              <a href=\\"http://blog\\">http://blog</a>
            </li>
            <li>
              <label style=\\"width:100px;margin-right:1.5rem\\">Youtube</label>
              <a href=\\"http://youtube\\">http://youtube</a>
            </li>
            <li>
              <label style=\\"width:100px;margin-right:1.5rem\\">Twitter</label>
              <a href=\\"http://twitter\\">http://twitter</a>
            </li>
            <li>
              <label style=\\"width:100px;margin-right:1.5rem\\">Reddit</label>
              <a href=\\"http://reddit\\">http://reddit</a>
            </li>
            <li>
              <label style=\\"width:100px;margin-right:1.5rem\\">Patreon</label>
              <a href=\\"http://patreon\\">http://patreon</a>
            </li>
          </ul>
        </div>
        <div>
          <h2>Bio</h2>
          <div class=\\"markdown\\">
            <p>Test user bio</p>
          </div>
        </div>
        <div>
          <h2>Self Authoring suite</h2>
          <ul>
            <li>Completed Past authoring</li>
            <li>Completed Present faults authoring</li>
            <li>Completed Present virtues authoring</li>
            <li>Completed Future authoring</li>
          </ul>
        </div>
        <div>
          <h2>Big 5</h2>
          <ul>
            <li>Completed Understand myself</li>
            <li>Agreeableness: 50
              <ul>
                <li>Compassion: 50</li>
                <li>Politeness: 50</li>
              </ul>
            </li>
            <li>Conscientiousness: 50
              <ul>
                <li>Industriousness: 50</li>
                <li>Orderliness: 50</li>
              </ul>
            </li>
            <li>Extraversion: 50
              <ul>
                <li>Enthusiasm: 50</li>
                <li>Orderliness: 50</li>
              </ul>
            </li>
            <li>Neuroticism: 50
              <ul>
                <li>Withdrawal: 50</li>
                <li>Volatility: 50</li>
              </ul>
            </li>
            <li>Openness to experience: 50
              <ul>
                <li>Intellect: 50</li>
                <li>Openness: 50</li>
              </ul>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <footer>
      <div class=\\"container\\">
        <hr>
        <p>Brought to you with &#x2764; by
          <a href=\\"/about\\">Nick Redmark</a>. Support this platform on
          <a href=\\"https://www.patreon.com/nickredmark\\" target=\\"_blank\\">Patreon</a>.</p>
      </div>
    </footer>
  </div>
</div>"
`;

exports[`user displays reads 1`] = `
"<div data-reactroot=\\"\\">
  <div>
    <div style=\\"padding-top:50px;padding-bottom:20px\\">
      <nav class=\\"navbar navbar-inverse navbar-fixed-top\\">
        <div class=\\"container\\">
          <div class=\\"navbar-header\\">
            <button type=\\"button\\" class=\\"navbar-toggle collapsed\\" data-toggle=\\"collapse\\" data-target=\\"#navbar\\" aria-expanded=\\"false\\" aria-controls=\\"navbar\\"><span class=\\"sr-only\\">Toggle navigation</span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span></button>
            <a class=\\"navbar-brand\\" href=\\"/\\">Sorters Club</a>
          </div>
          <div id=\\"navbar\\" class=\\"collapse navbar-collapse\\">
            <ul class=\\"nav navbar-nav navbar-right\\">
              <li>
                <a href=\\"/\\">Home</a>
              </li>
              <li>
                <a href=\\"/about\\">About</a>
              </li>
              <li>
                <a href=\\"/users\\">Sorters</a>
              </li>
              <li class=\\"dropdownundefined\\">
                <a href=\\"#\\" class=\\"dropdown-toggle\\" data-toggle=\\"dropdown\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"true\\">Hello, test <span class=\\"caret\\"></span></a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
    <div class=\\"container\\">
      <div>
        <div>
          <h1>Test User</h1>
          <div class=\\"row\\">
            <div class=\\"col-xs-6 col-sm-3 col-md-2\\">
              <img alt=\\"Gravatar for undefined\\" style=\\"margin-bottom:1.5rem;width:100%;height:auto;margin-left:auto;margin-right:auto;display:block\\" src=\\"//www.gravatar.com/avatar/23463b99b62a72f26ed677cc556c44e8?d=retro&amp;r=g&amp;s=200\\" srcset=\\"//www.gravatar.com/avatar/23463b99b62a72f26ed677cc556c44e8?d=retro&amp;r=g&amp;s=400 2x\\"
                height=\\"200\\" width=\\"200\\" class=\\"react-gravatar\\">
            </div>
            <div class=\\"col-xs-12 col-sm-9 col-md-10\\">
              <p>
                <a href=\\"/u/test\\">/u/ test</a>
              </p>
              <div>
                <div class=\\"markdown\\">
                  <p>Test user about</p>
                </div>
              </div>
              <button class=\\"btn btn-primary\\">Send message</button>
            </div>
          </div>
          <ul class=\\"nav nav-tabs\\">
            <li role=\\"presentation\\">
              <a href=\\"/u/test\\">&#x26CF; Activity</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/profile\\">&#x1F464; Profile</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/goals\\">&#x25CE; Goals</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/journal\\">&#x270E; Journal</a>
            </li>
            <li role=\\"presentation\\" class=\\"active\\">
              <a href=\\"/u/test/reads\\">&#x1F4D6; Books</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/topics\\">&#x1F4A1; Topics</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/essays\\">&#x270E; Essays</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/speeches\\">&#x1F444; Speeches</a>
            </li>
            <li role=\\"presentation\\">
              <a href=\\"/u/test/conversations\\">&#x1F5E9; Conversations</a>
            </li>
          </ul>
        </div>
        <h2>Reads</h2>
        <div class=\\"markdown\\">
          <p>Test user reading</p>
        </div>
        <ul>
          <li>
            <a href=\\"/read/<id>\\">Read</a>
            <ul></ul>
          </li>
          <li>
            <a href=\\"/read/<id>\\">Read - reading</a>
            <ul></ul>
          </li>
          <li>
            <a href=\\"/read/<id>\\">Read - read</a>
            <ul></ul>
          </li>
        </ul>
      </div>
    </div>
    <footer>
      <div class=\\"container\\">
        <hr>
        <p>Brought to you with &#x2764; by
          <a href=\\"/about\\">Nick Redmark</a>. Support this platform on
          <a href=\\"https://www.patreon.com/nickredmark\\" target=\\"_blank\\">Patreon</a>.</p>
      </div>
    </footer>
  </div>
</div>"
`;
